docs/docs/javascript/overview/eslint.mdx

```diff
@@ -0,0 +1,554 @@
+    "--format=json",
+    "--output-file=reports/eslint.json",
+    "--cache",
+    "--max-warnings=0",
+    "--no-error-on-unmatched-pattern"
+]
+json_output = true  # Enables summary parsing in console
+```
+
+**Development configuration with detailed reporting:**
+```toml title="pants.toml"
+[eslint]
+args = [
+    "--cache",
+    "--report-unused-disable-directives",
+    "--format=stylish"
+]
+```
+
+:::tip Cache Location
+ESLint will create a `.eslintcache` file to store cache data. Consider adding this to your `.gitignore`:
+
+```gitignore title=".gitignore"
+.eslintcache
+reports/
+```
+:::
+
+### Common Issues
+
+1. **ESLint not finding config files**: Ensure config files are in the project root or use explicit config path with `config = "path/to/eslint.config.js"`
+2. **TypeScript errors**: Install TypeScript ESLint plugins using `extra_dev_dependencies` and configure parser in your ESLint config
+3. **Performance issues**: Enable ESLint caching with `args = ["--cache"]` for faster subsequent runs
+4. **Cache invalidation**: Delete `.eslintcache` if you encounter stale results after config changes
+5. **Build failures on warnings**: Use `args = ["--max-warnings=0"]` to fail builds on any warnings, or `args = ["--quiet"]` to suppress warnings
+6. **Empty file sets**: Use `args = ["--no-error-on-unmatched-pattern"]` to prevent failures when no files match patterns
+7. **Output format issues**: Use `json_output = true` for parsed summaries, or `args = ["--format=json"]` for raw JSON output
+
+### Debug Mode
+
+Enable debug logging to troubleshoot issues:
+
+```bash
+$ pants --level=debug lint src/js/::
+```
+
+### Custom ESLint Rules
+
+To use custom ESLint rules or plugins, ensure they're installed in your project:
+
+```json title="package.json"
+{
+  "devDependencies": {
+    "eslint-plugin-custom": "^1.0.0"
+  }
+}
+```
+
+## Examples
+
+### Basic JavaScript Project
+
+```
+my-project/
+├── package.json
+├── .eslintrc.json
+├── src/
+│   ├── BUILD
+│   ├── main.js
+│   └── utils.js
+└── pants.toml
+```
+
+```toml title="pants.toml"
+[GLOBAL]
+backend_packages.add = [
+  "pants.backend.javascript",
+  "pants.backend.experimental.javascript.lint.eslint",
+]
+```
+
+```python title="src/BUILD"
+js_sources(
+    name="src",
+    sources=["*.js"],
+)
+```
+
+### TypeScript Project
+
+```
+my-project/
+├── package.json
+├── tsconfig.json
+├── eslint.config.js
+├── src/
+│   ├── BUILD
+│   ├── main.ts
+│   ├── types.ts
+│   └── utils.ts
+└── pants.toml
+```
+
+```toml title="pants.toml"
+[GLOBAL]
+backend_packages.add = [
+  "pants.backend.javascript",
+  "pants.backend.typescript",
+  "pants.backend.experimental.javascript.lint.eslint",
+]
+
+[eslint]
+extra_dev_dependencies = [
+    "@typescript-eslint/parser@^6.0.0",
+    "@typescript-eslint/eslint-plugin@^6.0.0"
+]
+```
+
+```python title="src/BUILD"
+typescript_sources(
+    name="backend",
+    sources=["*.ts"],
+)
+```
+
+
+### Mixed JavaScript/TypeScript Project
+
+```
+my-project/
```

Comment:
Are multiple, independent projects supported? E.g. my-project and my-project-v2 with different extra-dev-dependencies?